### PR TITLE
[PLAT-71332] Extract the metric name from any expensive query

### PR DESCRIPTION
**Pull request by hczhu**
# How to test

```
$ go test -run '.*ExtractMetricName.*' github.com/m3db/m3/src/query/api/v1/handler/prom
ok      github.com/m3db/m3/src/query/api/v1/handler/prom        (cached)

```
# Why?
The current query label of the metric for expensive queries are unreadable because M3 sanitize the label strings.
See the picture below.
image

### CODE REVIEW

File: src/query/api/v1/handler/prom/read.go [hunk 1]
```diff
@@ -249,6 +252,36 @@ func (h *readHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// NB: this is a naive but lightweight method to extra a metric name from a PromQL query.
+// It returns an empty string if it fails to extract a metric name.
+// We don't want to parse the PromQL here because the extraction is not super important.
+func (h *readHandler) extractMetricName(query string) string {
```
Comment (jnyi) [inline src/query/api/v1/handler/prom/read.go:258]: This only extracts the first metric name while query could be a combination of multiple metrics.

**Review by jnyi — APPROVED** [review]
LGTM since it only affects large queries.
 
I think the sanitize is done by @davidyuanfs in this method `truncateQuery`.

File: src/query/api/v1/handler/prom/read.go (hunk 1 above)
Comment (hczhu) [reply src/query/api/v1/handler/prom/read.go:258]: I know. It should help the majority cases.

**Comment by hczhu** [comment]
> LGTM since it only affects large queries.
> 
> I think the sanitize is done by @davidyuanfs in this method `truncateQuery`.



> LGTM since it only affects large queries.
> 
> I think the sanitize is done by @davidyuanfs in this method `truncateQuery`.

Metric label sanitization is done by `uber-go/tally`. Otherwise, I could have modified the sanitization function. `truncateQuery` is a trivial function. 
https://src.dev.databricks.com/databricks/m3/-/blob/src/query/api/v1/handler/prom/read.go?L252

**Pull request merged**